### PR TITLE
Handle case where malicious URLs start with %20

### PR DESCRIPTION
**Pull request by skunkworks**

### CODE REVIEW

**Comment by EvanHahn** [comment]
Should we filter anything with `javascript:`? Or could we be stricter about the protocols we allow?

**Comment by skunkworks** [comment]
We aren't able to be stricter with the protocols because "some internal apps" use `sanitize-url` to handle mobile app deep link URLs.

It does seem slightly safer to remove anything with `javascript:`, e.g. just make the regex `*javascript:*`. I'll go with that.

**Comment by crookedneighbor** [comment]
Not sure we want to make the regex `.*javascript:.*. `.

I think including a `:` after a `#` can be a valid character. So `http://example.com#myjavascript:foo` is _technically_ a valid url, but would be converted to `about:blank`. It's definitely an edge case, but one I think we should account for.

**Comment by EvanHahn** [comment]
Could you parse out the protocol and run the regular expression on that? You could also probably do this with a different regex.

File: index.js [hunk 1]
```diff
@@ -1,15 +1,25 @@
 'use strict';
 
-var jsRegex = /^(%20)*javascript:.*/im;
-var dataRegex = /^(%20)*data:.*/im;
+var invalidPrototcolRegex = /^(%20)*(javascript|data)/im;
 var ctrlCharactersRegex = /[^\x20-\x7E]/gmi;
+var urlSchemeRegex = /^(.+):/gm;
```
Comment (EvanHahn) [inline index.js]: Pretty sure this regex is wrong.

```js
> 'http://example.com:3000'.match(urlSchemeRegex)
// [ 'http://example.com:' ]

> 'http://example.com/path/to:something'.match(urlSchemeRegex)
// [ 'http://example.com/path/to:' ]

> 'http://example.com/path/to:something:else'.match(urlSchemeRegex)
// [ 'http://example.com/path/to:something:' ]
```

Maybe this is desired, though.

I think you want the regex to be something like `/^([^:]+):/gm` or parse the protocol with a more robust method (like [using an `a` tag](https://evanhahn.com/parse-urls-with-a-tags/) or a [URL module](https://nodejs.org/api/url.html)).

Comment (crookedneighbor) [reply index.js]: Good call on the a tag, we'll do that.

Comment (crookedneighbor) [reply index.js]: Actually, we should be able to use this in a node context as well. So we can't use the `a` tag method.

**Pull request merged**